AutoRest/Generators/Extensions/Azure.Extensions/AzureExtensions.cs

```diff
@@ -214,7 +221,15 @@ public static void AddLongRunningOperations(ServiceClient serviceClient)
                     if (isLongRunning is bool && (bool)isLongRunning)
                     {
                         serviceClient.Methods.Insert(i, (Method) method.Clone());
-                        method.Name = "Begin" + method.Name.ToPascalCase();
+                        if (codeNamer != null)
+                        {
+                            method.Name = "Begin" + codeNamer.GetMethodName(method.Name).ToPascalCase();
```

Comment:
delete is a reserved word in nodejs. Hence methods named as "delete" are renamed as "deleteMethod" in generic nodejs. Cloning of delete to beginDelete in "Azure.NodeJS" happens before renaming of delete to deleteMethod in generic. beginDelete is not a reserved word hence does not get converted to beginDeleteMethod. Hence we make this change so that beginDelete is accurately coverted to beginDeleteMethod